napari/_qt/containers/qt_axis_model.py

```diff
@@ -0,0 +1,164 @@
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def __repr__(self) -> str:
+        return self.dims.axis_labels[self.axis]
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, int):
+            return self.axis == other
+        if isinstance(other, str):
+            return repr(self) == other
+        if isinstance(other, AxisModel):
+            return (self.dims is other.dims) and (self.axis == other.axis)
+        return NotImplemented
+
+    @property
+    def rollable(self) -> bool:
+        """
+        If the axis should be rollable.
+        """
+        return self.dims.rollable[self.axis]
+
+    @rollable.setter
+    def rollable(self, value: bool) -> None:
+        rollable = list(self.dims.rollable)
+        rollable[self.axis] = value
+        self.dims.rollable = tuple(rollable)
+
+
+class AxisList(SelectableEventedList[AxisModel]):
+    def __init__(self, axes: Iterable[AxisModel]):
+        super().__init__(axes)
+
+    @classmethod
+    def from_dims(cls, dims: Dims) -> 'AxisList':
+        """Create AxisList instance from Dims object.
+
+        The AxisList is filled with a number of AxisModels based
+        on the number of dimensions in the Dims object.
+
+        Parameters
+        ----------
+        dims : napari.components.dims.Dims
+            Dims object to be used for creation.
+
+        Returns
+        -------
+        AxisList
+            A selectable evented list of the viewer axes.
+        """
+        return cls([AxisModel(dims, dims.order[i]) for i in range(dims.ndim)])
+
+
+class QtAxisListModel(QtListModel[AxisModel]):
+    def data(self, index: QModelIndex, role: Qt.ItemDataRole):
+        if not index.isValid():
+            return None
+        axis = self.getItem(index)
+        if role == Qt.ItemDataRole.DisplayRole:
+            return str(axis)
+        if role == Qt.ItemDataRole.TextAlignmentRole:
+            return Qt.AlignCenter
+        if role == Qt.ItemDataRole.CheckStateRole:
+            return (
+                Qt.CheckState.Checked
+                if axis.rollable
+                else Qt.CheckState.Unchecked
+            )
+        return super().data(index, role)
+
+    def setData(
+        self,
+        index: QModelIndex,
+        value: Any,
+        role: int = Qt.ItemDataRole.EditRole,
+    ) -> bool:
+        axis = self.getItem(index)
+        if role == Qt.ItemDataRole.CheckStateRole:
+            axis.rollable = bool(
+                value == Qt.CheckState.Checked or value is True
+            )
+        elif role == Qt.ItemDataRole.EditRole:
+            axis_labels = list(axis.dims.axis_labels)
+            axis_labels[axis.axis] = value
+            axis.dims.axis_labels = tuple(axis_labels)
+        else:
+            return super().setData(index, value, role=role)
+        self.dataChanged.emit(index, index, [role])
+        return True
+
+    def flags(self, index: QModelIndex) -> Qt.ItemFlags:
```

Comment:
```suggestion
    def flags(self, index: QModelIndex) -> Qt.ItemFlag:
```
Without it this code crash on Qt6 6.5.2

Reply:
I think that's covered by the min qtpy version bump also in this PR.

See the latest successful run on [PyQt6 6.5.2](https://github.com/napari/napari/actions/runs/6124247526/job/16844775311?pr=5986#step:8:116).

Or there something about PySide6 6.5.2 that would break?